### PR TITLE
fix(core): bug fix for block table logic

### PR DESCRIPTION
**Pull request by rebel-jonghewk**
### 🚀 Summary of Changes

**Fix block-table range & simplify padding**
- Ensure block tables use IDs in 0..<estimated_block_size (regression from PR #49).
- Simplify padding:
  - Pad row size with -1.
  - Pad per-block entries with the first unused block ID (works whether a dummy block 0 exists or not).

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [x] 🛠 Bug fix (`bug-fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
3. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [x] PR title follows Conventional Commits format
* [x] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

File: vllm_rbln/model_executor/models/optimum/model_base.py [hunk 1]
```diff
@@ -216,35 +208,38 @@ def pad_decoder_items(
...
+            device=block_tables.device,
+        )
+        unused_blocks = available_blocks[
+            ~torch.isin(available_blocks, block_tables.flatten())]
+        block_num = block_tables.size(1)
+        mask = torch.ones(
+            (padded_batch_size, block_num),
+            dtype=torch.bool,
+            device=block_tables.device,
+        )
```
Comment (rebel-eunji) [inline vllm_rbln/model_executor/models/optimum/model_base.py:228]:
```suggestion
        mask = torch.ones_like(
            padded_block_tables,
            dtype=torch.bool,
            device=block_tables.device,
        )
```
```

File: vllm_rbln/worker/model_runner.py [hunk 2]
```diff
@@ -565,6 +565,7 @@ def execute_model(
                 model_input.attn_metadata.kv_caches = kv_caches
             for kv_cache in kv_caches:
                 self.compile_context.mark_static_address(kv_cache)
+
```
Comment (rebel-eunji) [inline vllm_rbln/worker/model_runner.py:568]: empty space
```suggestion
```

File: vllm_rbln/model_executor/models/optimum/model_base.py [hunk 3]
```diff
@@ -216,35 +208,38 @@ def pad_decoder_items(
                                           dtype=positions.dtype)
         padded_block_tables = torch.zeros(padded_batch_size,
                                           block_tables.shape[1],
-                                          dtype=block_tables.dtype).fill_(
-                                              self.padding_value)
+                                          dtype=block_tables.dtype).fill_(-1)
```
Comment (rebel-eunji) [inline vllm_rbln/model_executor/models/optimum/model_base.py:211]: I think fill_(-1) is not required. Because the column-wise padding is already filled with -1 in model_runner.
And the row-wise padding is also masked in below.
Is it the double padding for safety?

File: vllm_rbln/v1/worker/optimum_model_runner.py [hunk 4]
```diff
@@ -448,12 +453,16 @@ def _update_states(self, scheduler_output: "SchedulerOutput") -> None:
         new/resumed/paused/finished request in the batch.
         """
         for req_id in scheduler_output.finished_req_ids:
-            logger.debug(
-                "Request %s is finished. Prompt tokens: %s, "
-                "Generated tokens: %s, Freed block(s): %s", req_id,
-                len(self.requests[req_id].prompt_token_ids),
-                len(self.requests[req_id].output_token_ids),
-                self.requests[req_id].block_ids[0])
+            if logger.isEnabledFor(logging.DEBUG):
```
Comment (rebel-eunji) [inline vllm_rbln/v1/worker/optimum_model_runner.py:456]: I missed -1 in before logging. So I've fixed it in this PR.
@rebel-seinpark

File: vllm_rbln/model_executor/models/optimum/model_base.py (hunk 3 above)
Comment (rebel-jonghewk) [reply vllm_rbln/model_executor/models/optimum/model_base.py:213]: Yes. Why don't we keep it for safety?

**Review by rebel-eunji — APPROVED** [review]
LGTM 👍
